homeassistant/helpers/service.py

```diff
@@ -1268,7 +1268,16 @@ def async_register_entity_service(
     # the check could be extended to require All/Any to have sub schema(s)
     # with all entity service fields
     elif not cv.is_entity_service_schema(schema):
-        raise HomeAssistantError("The schema is not an entity service schema")
+        # pylint: disable-next=import-outside-toplevel
+        from .frame import report
+
+        report(
+            (
+                "registers an entity service with a non entity service schema "
+                "which will stop working in HA Core 2025.10"
+            ),
+            error_if_core=False,
```

Comment:
Why don't we raise an error in core if already all services are correct?

Reply:
Core means "we did not find an integration when unwinding the stack frame", not "called by a core integration", so I'd rather not raise.